runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java

```diff
@@ -218,6 +226,74 @@ public void onWidgetLoaded(WidgetLoaded event)
 		}
 	}
 
+	@Subscribe
+	public void itemContainerChanged(ItemContainerChanged event)
+	{
+		final ItemContainer container = event.getItemContainer();
+
+		if (container == client.getItemContainer(InventoryID.INVENTORY))
+		{
+			final Item[] items = container.getItems();
+
+			long numberOfSupplyCrates = Arrays.stream(items)
+					.filter(item -> item.getId() == ItemID.SUPPLY_CRATE)
+					.count();
+
+			// We want to make sure that we actually opened the supply crate, not banked it
+			if (numberOfSupplyCrates != lastNumberOfSupplyCrates && inventorySnapshot != null)
+			{
+				log.debug("Number of supply crates in inventory has changed from {} to {}", lastNumberOfSupplyCrates, numberOfSupplyCrates);
+
+				// we lost a crate by opening it
+				if (numberOfSupplyCrates < lastNumberOfSupplyCrates && !bankWidgetIsOpen())
+				{
+					List<Item> loot = itemDelta(inventorySnapshot, items);
+
+					final Collection<ItemStack> lootStacks = loot.stream()
+							.map(item -> new ItemStack(item.getId(), item.getQuantity()))
+							.collect(Collectors.toList());
+
+					if (!lootStacks.isEmpty())
```

Comment:
If you're correctly detecting the event this will never be empty, meaning you can remove this if to reduce cognitive complexity.